changelog/2537.added.md

```diff
@@ -0,0 +1 @@
+Add support for `syslog`, `openlog`, `closelog` on `macos`.
```

Comment:
Looks like `closelog()` is not included in the PR

Reply:
Weird .. Perhaps get reverted in the setlogmark func. Let me add it back.